client/src/components/PhotoModal.jsx

```diff
@@ -0,0 +1,128 @@
+      this.setState({
+        currentPhoto: this.state.currentPhoto -= 1,
+      });
+    } else {
+      this.setState({
+        currentPhoto: this.state.currentPhoto = this.props.photoDetails.length - 1,
+      });
+    }
+  }
+
+  handleRightArrowClick() {
+    if (this.state.currentPhoto < this.props.photoDetails.length - 1) {
+      this.setState({
+        currentPhoto: this.state.currentPhoto += 1,
+      });
+    } else {
+      this.setState({
+        currentPhoto: this.state.currentPhoto = 0,
+      });
+    }
+
+  }
+
+  render() {
+
+    return (
+      <div>
+        <div className="photoModuleParentContainer">
+          <div className="photoModuleLeftContainer">
+            <div className="photoModuleLeftArrowContainer">
+              <div className="photoModuleLeftArrow">
+                <span className="FaChevronLeft" onClick={this.handleLeftArrowClick}> <FaChevronLeft /> </span>
+              </div> {/*photoModuleLeftArrow*/}
+
+            </div> {/*photoModuleLeftArrowContainer*/}
+
+            <div className="photoModuleMainPhotoContainer">
+              <div className="photoModuleMainPhoto">
+                <Photo photoClass={'modalPhoto'} photoUrl={this.props.photoDetails[this.state.currentPhoto]}/>
+
+              </div> {/*photoModuleMainPhoto*/}
+
+            </div> {/*photoModuleMainPhotoContainer*/}
+
+            <div className="photoModuleRightArrowContainer">
+              <div className="photoModuleRightArrow">
+                <span className="FaChevronRight"  onClick={this.handleRightArrowClick}> <FaChevronRight /> </span>
+
+              </div> {/*photoModuleRightArrow*/}
+
+            </div> {/*photoModuleRightArrowContainer*/}
+
+          </div> {/*photoModuleLeftContainer*/}
+
+          <div className="photoModuleRightContainer">
+            <div className="photoModuleExitButtonContainer">
+              <div className="photoModuleExitButton" onClick={this.props.viewPhotoButtonClick}>
+                <span className="FaTimes"> <FaTimes /> </span>
+              </div> {/*photoModuleExitButton*/}
+
+            </div> {/*photoModuleExitButtonContainer*/}
+
+            <div className="photoModalCarouselContainer">
+              <PhotoModalCarousel photos={this.props.photoDetails} currentPhoto={this.state.currentPhoto}/>
+
+            </div> {/*photoModalCarouselContainer*/}
+
+            <PhotoModalDescriptionContainer photos={this.props.photoDetails} currentPhoto={this.state.currentPhoto}/>
+
+          </div> {/*photoModuleRightContainer*/}
+
+        </div> {/*photoModuleParentContainer*/}
+
+      </div> /*starting div*/
+    )
+  }
+}
+
+const PhotoModalCarousel = (props) => {
+  let carouselPhotos = props.photos.map((photo, i) => {
+    if (i === props.currentPhoto) {
+      return <Photo photoClass={'carouselPhoto'} photoUrl={photo} key={i}/>
+    } else {
+      return <Photo photoClass={'carouselPhotoActive'} photoUrl={photo} key={i}/>
+    }
+  })
```

Comment:
Do all of this on line 107. There is nothing conditional about this view, so no need to create a variable to store the dynamically rendered list of photos.